src/components/Reports/ReportEdit.tsx

```diff
@@ -1,6 +1,145 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import TfMain from 'src/components/common/TfMain';
+import PageForm from 'src/components/common/PageForm';
+import strings from 'src/strings';
+import ReportForm from 'src/components/Reports/ReportForm';
+import { Box, Typography, useTheme } from '@mui/material';
+import ReportService from 'src/services/ReportService';
+import { Report } from 'src/types/Report';
+import { useHistory, useParams } from 'react-router-dom';
+import { APP_PATHS } from 'src/constants';
+import ReportFormAnnual from 'src/components/Reports/ReportFormAnnual';
+import { Button, DialogBox } from '@terraware/web-components';
+import { FormButton } from 'src/components/common/FormBottomBar';
 
 export default function ReportEdit(): JSX.Element {
-  return <TfMain>Report Edit Page</TfMain>;
+  const { reportId } = useParams<{ reportId: string }>();
+  const reportIdInt = parseInt(reportId, 10);
+
+  const theme = useTheme();
+
+  const history = useHistory();
+
+  const [report, setReport] = useState<Report>();
+  useEffect(() => {
+    const getReport = async () => {
+      const result = await ReportService.getReport(reportIdInt);
+      if (result.requestSucceeded) {
+        setReport(result.report);
+      }
+    };
+
+    getReport();
+  }, [reportIdInt]);
+
+  const [showAnnual, setShowAnnual] = useState(false);
+
+  const [confirmSubmitDialogOpen, setConfirmSubmitDialogOpen] = useState(false);
+
+  const gotoReportView = async (saveChanges: boolean) => {
+    if (saveChanges && report) {
+      await ReportService.updateReport(report);
+    }
+
+    // unlock the report
+    await ReportService.unlockReport(reportIdInt);
```

Comment:
Should we check if ReportService.updateReport succedded before unlocking?

Reply:
Yes, I will have to add checking the response to service calls in many places.